Cargo.toml

```diff
@@ -16,4 +16,10 @@ keywords = ["data-structures"]
 readme = "README.md"
 
 [features]
-nightly = []
+nightly = ["serde"]
+
+[dependencies]
+serde = { optional = true, version = "^0.7" }
+
+[dev-dependencies]
+serde_json = "^0.7"
```

Comment:
Unfortunately, this causes `serde` and `serde_json` to be downloaded and compiled even if the user doesn't enable the `serde` feature.